libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt

```diff
@@ -248,6 +250,18 @@ abstract class AbstractKotlinPlugin @Inject constructor(val scriptHandler: Scrip
             }
         })
     }
+
+    private fun configureResolutionStrategy(project: Project) {
+        val projectVersion = loadKotlinVersionFromResource(project.logger)
+        project.configurations.all(Action<Configuration> { configuration ->
+            configuration.resolutionStrategy.eachDependency { details ->
+                val requested = details.requested
+                if (requested.group.equals("org.jetbrains.kotlin") && requested.version.isEmpty()) {
+                    details.useTarget("${requested.group}:${requested.name}:$projectVersion")
```

Comment:
Why do you put `project.version` here? shouldn't you somehow lookup Kotlin plugin version here?